checkov/secrets/plugins/custom_regex_detector.py

```diff
@@ -43,26 +50,85 @@ def analyze_line(
+
+        self._find_potential_secret(
+            filename=filename,
+            string_to_analyze=line,
+            output=output,
+            line_number=line_number,
+            context=raw_context,
+            is_multiline=False,
+            **kwargs
+        )
+
+        if filename not in self._analyzed_files:
+            self._analyzed_files.add(filename)
+            file_content = None
+            try:
+                with open(filename, 'r') as f:
+                    file_content = f.read()
+            except Exception:
+                logging.warning(
+                    "Could not open file in order to detect secrets}",
+                    extra={"file_path": filename}
+                )
+            if not file_content:
+                return output
+
+            self._find_potential_secret(
+                filename=filename,
+                string_to_analyze=file_content,
+                output=output,
+                line_number=line_number,
+                context=raw_context,
+                is_multiline=True,
+                **kwargs
+            )
+
+        return output
+
+    def _find_potential_secret(
+            self,
+            filename: str,
+            string_to_analyze: str,
+            output: Set[PotentialSecret],
+            line_number: int = 0,
+            context: Optional[CodeSnippet] = None,
+            is_multiline: bool = False,
+            **kwargs: Any
+    ) -> None:
+        current_denylist: Set[Pattern[str]] = self.multiline_deny_list if is_multiline else self.denylist
+        current_regex_to_metadata: dict[str, dict[str, Any]] = self.multiline_regex_to_metadata if is_multiline else self.regex_to_metadata
+        kwargs["regex_denylist"] = current_denylist
+        for match, regex in self.analyze_string(string_to_analyze, **kwargs):
             try:
                 verified_result = call_function_with_arguments(self.verify, secret=match, context=context)
                 is_verified = True if verified_result == VerifiedResult.VERIFIED_TRUE else False
             except Exception:
                 is_verified = False
-            regex_data = self.regex_to_metadata[regex.pattern]
-            ps = PotentialSecret(type=regex_data["Name"], filename=filename, secret=match,
-                                 line_number=line_number, is_verified=is_verified)
-            ps.check_id = self.regex_to_metadata[regex.pattern]["Check_ID"]  # type:ignore[attr-defined]
-            if len(cast(str, ps.secret_value)) in range(MIN_CHARACTERS, MAX_CHARACTERS) or not regex_data['isCustom']:
+            regex_data = current_regex_to_metadata[regex.pattern]
+            # Wrap multiline match with fstring + ''
+            match = f"'{match}'" if is_multiline else match
+            ps = PotentialSecret(
+                type=regex_data["Name"],
+                filename=filename,
+                secret=match,
+                line_number=line_number,
```

Comment:
If we're scanning the entire file, what line_number will be shown in case of a match?

Reply:
yup, I don't think it's relevant as it will indicate the error any way. Do you think it should be changed to 0 or something like that ?

Reply:
Yes default of zero for multiline match sounds reasonable